src/sources/fluent/mod.rs

```diff
@@ -92,8 +101,50 @@ impl SourceConfig for FluentConfig {
         )
     }
 
-    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
-        vec![Output::default(DataType::Log)]
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        // `host_key` is only inserted if not present already.
+        let host_key = parse_value_path(log_schema().host_key())
+            .ok()
+            .map(LegacyKey::InsertIfEmpty);
+
+        let tag_key = parse_value_path("tag").ok().map(LegacyKey::Overwrite);
+
+        let log_namespace = global_log_namespace.merge(self.log_namespace);
+
+        // There is a global and per-source `log_namespace` config.
+        // The source config overrides the global setting and is merged here.
+        let mut schema_definition = BytesDeserializerConfig
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata()
+            .with_source_metadata(
+                FluentConfig::NAME,
+                host_key,
+                &owned_value_path!("host"),
+                Kind::bytes().or_undefined(),
+                Some("host"),
+            )
+            .with_source_metadata(
+                FluentConfig::NAME,
+                tag_key,
+                &owned_value_path!("tag"),
+                Kind::bytes().or_undefined(),
```

Comment:
I missed these `_or*` functions in other PRs. Good catching finding them.

Reply:
:+1:  glad you asked the question about these, because turns out I had not correct usage here, will update the diff.